qemu/tests/block_commit_reboot.py

```diff
@@ -0,0 +1,47 @@
+import time
+import random
+import logging
+
+from qemu.tests import blk_commit
+
+
+class BlockCommitReboot(blk_commit.BlockCommit):
+
+    def reboot(self):
+        """
+        Reset guest with system_reset;
+        """
+        return super(BlockCommitReboot, self).reboot(boot_check=False)
+
+    def action_when_start(self):
+        """
+        start pre-action in new threads;
+        """
+        super(BlockCommitReboot, self).action_when_start()
+        time.sleep(random.randint(0, 20))
+
+
+def run(test, params, env):
+    """
+    block_commit_reboot test:
+    1). boot a guest, then reboot the guest with system_reset;
+    2). create snapshots and start commit job immediately;
+    3). waiting commit done and check guest is alive;
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    tag = params.get("source_image", "image1")
+    reboot_test = BlockCommitReboot(test, params, env, tag)
+    try:
+        reboot_test.action_when_start()
+        reboot_test.create_snapshots()
+        reboot_test.start()
+        reboot_test.action_after_finished()
+    finally:
+        try:
+            reboot_test.clean()
+        except Exception, e:
+            logging.warn(e)
+            pass
```

Comment:
This line is useless.

Reply:
@luckyh  Updated. Please help review again. Thanks.